src/tests/test_magic_cte.py

```diff
@@ -0,0 +1,34 @@
+def test_trailing_semicolons_removed_from_cte(ip):
+    ip.run_cell(
+        """%%sql --save positive_x
+SELECT * FROM number_table WHERE x > 0;
+
+
+"""
+    )
+
+    ip.run_cell(
+        """%%sql --save positive_y
+SELECT * FROM number_table WHERE y > 0;
+"""
+    )
+
+    cell_execution = ip.run_cell(
+        """%%sql --save final --with positive_x --with positive_y
```

Comment:
We need to track this in [CTE followup issue](https://github.com/ploomber/jupysql/issues/512)